packages/snaps-simulator/src/features/builder/utils.ts

```diff
@@ -35,10 +38,82 @@ export function getNodeText(nodeModel: NodeModel<Component>) {
  * @param node - The node to verify.
  * @returns True if the node is a valid form children, otherwise false.
  */
-export function isValidFormNode(node: Component) {
-  return node.type === 'input' || node.type === 'button';
+export function isValidFormNode(node: JSXElement) {
+  return node.type === 'Field' || node.type === 'Button';
+}
+
+/**
+ * Verify that the node is a valid field children.
+ *
+ * @param node - The node to verify.
+ * @param field - The field element.
+ * @returns True if the node is a valid field children, otherwise false.
+ */
+export function isValidFieldNode(
+  node: JSXElement,
+  field: FieldElement,
+): node is InputElement | ButtonElement {
+  const fieldChildren = getJsxChildren(field);
+
+  if (
+    (fieldChildren as JSXElement[]).some(
+      (child) =>
+        (child.type === 'Button' && node.type === 'Button') ||
+        (child.type === 'Input' && node.type === 'Input'),
+    )
+  ) {
+    return false;
+  }
+
+  return node.type === 'Input' || node.type === 'Button';
```

Comment:
Fields can also take dropdowns and soon file inputs. Can we make this more generic somehow?